### PR TITLE
Fix mypy issues in www/views.py

### PR DESCRIPTION
**Pull request by ephraimbuddy**
This PR fixes the mypy issues in views.py

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/www/views.py [hunk 1]
```diff
@@ -30,12 +30,13 @@
...
 from operator import itemgetter
-from typing import Any, Callable, List, Optional, Set, Tuple, Union
+from typing import Any, Callable, Dict, List, Optional, Set, Tuple, Union
 from urllib.parse import parse_qsl, unquote, urlencode, urlparse
 
 import lazy_object_proxy
 import markupsafe
 import nvd3
+import sqlalchemy
 import sqlalchemy as sqla
```
Comment (uranusjr) [inline airflow/www/views.py]: There’s already an alias imported so you can use `sqla` for where you’re using `sqlalchemy` and onmit the import addition.

File: airflow/www/views.py [hunk 2]
```diff
@@ -543,7 +544,8 @@ class AirflowBaseView(BaseView):
     }
 
     if not conf.getboolean('core', 'unit_test_mode'):
-        extra_args['sqlite_warning'] = settings.Session.bind.dialect.name == 'sqlite'
+        session: sqlalchemy.orm.session.Session = settings.Session
+        extra_args['sqlite_warning'] = session.bind.dialect.name == 'sqlite'
```
Comment (uranusjr) [inline airflow/www/views.py]: Although it is likely better to fix this in `settings` instead since I’d expect this very issue to pop up for other modules.

I think changing `Session: Optional[SASession] = None` there to just `Session: SASession` would fix the issue.

Comment (ephraimbuddy) [reply airflow/www/views.py]: Ok. I will try it

**Review by uranusjr — APPROVED** [review]

**Comment by potiuk** [comment]
Needs conflict resolution @ephraimbuddy

**Comment by ephraimbuddy** [comment]
Done. @uranusjr you can also take another look as I changed what you recently merged

File: airflow/settings.py [hunk 3]
```diff
@@ -80,7 +80,7 @@
 DAGS_FOLDER: str = os.path.expanduser(conf.get('core', 'DAGS_FOLDER'))
 
 engine: Optional[Engine] = None
-Session: Callable[..., SASession]
+Session: SASession
```
Comment (uranusjr) [inline airflow/settings.py]: Let’s use the `Callable` form (it’s done after a discussion in another PR)

Comment (ephraimbuddy) [reply airflow/settings.py]: mypy is complaining for callable:
```
airflow/www/views.py:546: error: "Callable[..., Any]" has no
attribute "bind"
    ... extra_args['sqlite_warning'] = settings.Session.bind...
                                       ^
Found 1 error in 1 file (checked 1 source file)
```

Comment (uranusjr) [reply airflow/settings.py]: I guess we need my new fancy toy then!

```python
class SessionMaker(Protocol):
    """Fake the type returned by sessionmaker for typg hints.

    Add more things to this class as we need them.
    """

    bind: Engine

    def __call__(self, **kwargs: Any) -> SASession:
        ...

    def remove(self) -> None:
        ...


Session: SessionMaker
```

Comment (ashb) [reply airflow/settings.py]: Or change usages of `settings.Session.bind` to `settings.engine`

Comment (uranusjr) [reply airflow/settings.py]: The downside of `settings.engine` is it’s an `Optional`. Pick your poision I guess :p

Comment (ephraimbuddy) [reply airflow/settings.py]: Nice. Updated with settings.engine

File: airflow/www/views.py [hunk 4]
```diff
@@ -543,7 +543,7 @@ class AirflowBaseView(BaseView):
     }
 
     if not conf.getboolean('core', 'unit_test_mode'):
-        extra_args['sqlite_warning'] = settings.Session.bind.dialect.name == 'sqlite'
+        extra_args['sqlite_warning'] = settings.engine.bind.dialect.name == 'sqlite'
```
Comment (uranusjr) [inline airflow/www/views.py]:
```suggestion
        extra_args['sqlite_warning'] = settings.engine.dialect.name == 'sqlite'
```

I think Mypy will complain None does not have attribute dialect though. Consider the protocol approach, please?

Comment (ephraimbuddy) [reply airflow/www/views.py]: The engine is no longer optional. A change made in another PR, so let's see what happens

**Comment by ephraimbuddy** [comment]
This looks good now. Let me know if I should merge. @uranusjr

**Comment by uranusjr** [comment]
There's a test in the AWS provider failing, not sure if it's related to the AWS outage today. Likely not related though.

**Review by uranusjr — APPROVED** [review]

File: airflow/www/views.py (hunk 4 above)
Comment (ashb) [reply airflow/www/views.py]: Yeah, I made it non-optional in my models.dag mypy PR.
